pkg/kubectl/cmd/run.go

```diff
@@ -276,14 +276,14 @@ func RunRun(f cmdutil.Factory, cmdIn io.Reader, cmdOut, cmdErr io.Writer, cmd *c
 
 	params["env"] = cmdutil.GetFlagStringArray(cmd, "env")
 
-	allErrs := []error{}
-	var runObjectMap = map[string]*RunObject{}
+	var createdObjects = []*RunObject{}
 	runObject, err := createGeneratedObject(f, cmd, generator, names, params, cmdutil.GetFlagString(cmd, "overrides"), namespace)
 	if err != nil {
-		allErrs = append(allErrs, err)
+		return err
 	} else {
-		runObjectMap[generatorName] = runObject
```

Comment:
The key should be `serviceGenerator`?  
cc @xiangpengzhao

Reply:
I found the key is not used at all,  so I changed it to a slice

Reply:
@janetkuo yeah it should be `serviceGenerator` and using `serviceGenerator` fixes the second issue. However, as @mountkin mentioned, the key is not used at all. So it's safe to change it to a slice and it does fix the second issue as well.